packages/phoenix-client/src/phoenix/client/resources/spans/__init__.py

```diff
@@ -86,6 +89,18 @@ def get_spans_dataframe(
 
             _ = pd  # Prevent unused symbol error
 
+            if project_identifier and project_name:
+                raise ValueError("Provide only one of 'project_identifier' or 'project_name'.")
+            elif project_identifier and not project_name:
+                project_response = self._client.get(
+                    url=f"v1/projects/{project_identifier}",
+                    headers={"accept": "application/json"},
+                    timeout=timeout,
+                )
```

Comment:
Given a simple heuristic you can determine if this is an identifier (e.g. if it's a base64 encoded string or not). In addition if the string is encodable in the URL we don't actually have to resolve the project. I would probably add this code sooner rather than later since it reduces network calls. I think it exists somewhere.